compiler/rustc_mir_transform/src/sroa.rs

```diff
@@ -48,11 +54,42 @@ impl<'tcx> MirPass<'tcx> for ScalarReplacementOfAggregates {
 /// - the locals is a union or an enum;
 /// - the local's address is taken, and thus the relative addresses of the fields are observable to
 ///   client code.
-fn escaping_locals(excluded: &BitSet<Local>, body: &Body<'_>) -> BitSet<Local> {
+fn escaping_locals<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    excluded: &BitSet<Local>,
+    body: &Body<'tcx>,
+) -> BitSet<Local> {
+    let is_excluded_ty = |ty: Ty<'tcx>| {
+        if ty.is_union() || ty.is_enum() {
+            return true;
+        }
+        if let ty::Adt(def, _substs) = ty.kind() {
+            if def.variants().len() != 1 {
+                // ADTs with more than one field are fine to decompose
```

Comment:
This check is about multiple variants, so an enum. Is this even reachable? The comment also belongs on the next condition

Reply:
🤔 Clearly I misunderstood the documentation. I was trying to exclude structs with one field

Reply:
Ah yeah, I see now.